example/llmapper/app.py

```diff
@@ -0,0 +1,81 @@
+import streamlit as st
+from core import LLMResponse, create_knowledge_graph, draw_graph, read_summarize_prompt
+
+import promptulate as pne
+from promptulate.tools.wikipedia import wikipedia_search
+
+# Create a sidebar to place the user parameter configuration
+with st.sidebar:
+    model_name: str = st.selectbox(
+        label="Language Model Name",
+        options=[
+            "openai/gpt-4o",
+            "openai/gpt-4-turbo",
+            "deepseek/deepseek-chat",
+            "zhipu/glm-4",
+            "ollama/llama2",
+        ],
+        help="For more details, please see"
+        "[how to write model name?]("
+        "https://www.promptulate.cn/#/other/how_to_write_model_name)",
+    )
+    api_key = st.text_input("API Key", key="provider_api_key", type="password")
+    api_base = st.text_input("OpenAI Proxy URL (Optional)")
+    "[View the source code](https://github.com/Undertone0809/promptulate)"
+
+# Set title
+st.title("💬 Chat")
+st.caption(
+    "🚀 Hi there! 👋 I am a chatbot to draw knowledge map by Promptulate to help you."
+)
+# Determine whether to initialize the message variable
+# otherwise initialize a message dictionary
+if "messages" not in st.session_state:
+    st.session_state["messages"] = [
+        {"role": "assistant", "content": "How can I help you?"},
+    ]
+# Traverse messages in session state
+for msg in st.session_state.messages:
+    st.chat_message(msg["role"]).write(msg["content"])
+
+# User input
+if prompt := st.chat_input():
+    if not api_key:
+        st.info("Please add your API key to continue.")
+        st.stop()
+    # Display the user's message
+    st.session_state.messages.append({"role": "user", "content": prompt})
+    st.chat_message("user").write(prompt)
+
+    query_result: str = wikipedia_search(prompt, top_k_results=1)
+    # display search results in the chatbot panel
+    # st.chat_message("assistant").write(query_result)
+
+    summarize_prompt_path = "prompt-summarize.md"
```

Comment:
Simply use variables directly, do not take a separate markdown file to do reading.